src/System.Memory/src/System/SpanHelpers.cs

```diff
@@ -52,7 +52,36 @@ public static IntPtr Add<T>(this IntPtr start, int index)
         /// <summary>
         /// Determine if a type is eligible for storage in unmanaged memory. TODO: To be replaced by a ContainsReference() api.
         /// </summary>
-        public static bool IsReferenceFree<T>() => PerTypeLatches<T>.IsReferenceFree;
+        public static bool IsReferenceFree<T>()
+        {
+            // Under the JIT, these become constant-folded.
```

Comment:
Well we're in the "one-time-only" path at that point so it's hard to measure (except by turning off the latching but then that's not an interesting measure...) 

What are we angling for here? We already have the Type in hand, GetTypeInfo() is an identity function (and hopefully will go away entirely as migrate away from our 1.0 refactoring), and so we'd just be short-cutting IsPrimitive() with our own ladder of comparisons. Are we betting on that being faster than IsPrimitive()?